client/src/containers/AuthContainer.test.tsx

```diff
@@ -98,3 +88,122 @@ describe("verifying valid input for signup page", () => {
     });
   });
 });
+
+describe("signUp component", () => {
+  describe("handleSubmit function", () => {
+    it("should show password mismatch alert", async () => {
+      screen.getByTestId("SignUpTabButton").click();
+
+      const fields = [
+        ["passwordForm", "password"],
+        ["confirmPassForm", "passwordd"],
+      ];
+      fields.forEach((pair) => {
+        fireEvent.change(
+          screen
+            .getByTestId(pair[0])
+            .querySelector("input") as HTMLInputElement,
+          { target: { value: pair[1] } }
+        );
+      });
+
+      const form = screen.getByTestId("form");
+      const mockAlert = jest
+        .spyOn(window, "alert")
+        .mockImplementation(() => {});
+
+      fireEvent.submit(form);
+      expect(mockAlert).toBeCalledWith("Passwords do not match.");
+    });
+
+    it("should show backend error alert on backend error", async () => {
+      screen.getByTestId("SignUpTabButton").click();
+
+      const form = screen.getByTestId("form");
+      const msg = "test";
+      const mockSignUp = jest
+        .spyOn(ServerApi.prototype, "signUp")
+        .mockImplementation(() =>
+          Promise.resolve({
+            error: { response: { data: { message: msg } } } as AxiosError,
+          })
+        );
+      const mockAlert = jest
+        .spyOn(window, "alert")
+        .mockImplementation(() => {});
+
+      fireEvent.submit(form);
+      /**
+       * Since fireEvent.submit is not async function we cannot use await
+       * Without an await, expect(mockAlert) assertion can run before
+       * fireEvent.submit can complete execution causing the test to fail.
+       * Therefore, we await on dummy Promise to delay execution while
+       * fireEvent.submit completes.
+       */
+      await new Promise((r) => setTimeout(r, 1));
+      expect(mockSignUp).toBeCalled();
+      expect(mockAlert).toBeCalledWith(`Message: ${msg}`);
+    });
+
+    it("should show email confirmation alert on success", async () => {
+      screen.getByTestId("SignUpTabButton").click();
+
+      const form = screen.getByTestId("form");
+      const mockSignUp = jest
+        .spyOn(ServerApi.prototype, "signUp")
+        .mockImplementation(() =>
+          Promise.resolve({ response: {} as AxiosResponse })
+        );
```

Comment:
This is perfect. What you could also do, 1/ is assert that it was called with the expected parameters. 2/ return some mock data then assert that is is displayed in the UI as expected (this is more relevant for the `posts` logic, but I'm leaving it here for reference).